src/consumer/engine.rs

```diff
@@ -43,7 +43,7 @@ pub struct ConsumerEngine<Exe: Executor> {
     event_rx: mpsc::UnboundedReceiver<EngineEvent<Exe>>,
     event_tx: UnboundedSender<EngineEvent<Exe>>,
     batch_size: u32,
-    remaining_messages: u32,
+    remaining_messages: i64,
```

Comment:
negative amount of messages are possible ?

Reply:
In the casual workflow, we use the `batch_size` as a maximum number that we could send using a send flow command message to ask the broker to send us messages, which is why we have a subtraction below (`batch_size` - `remaining_messages`).

In the special case of batch messages (which is defined by clients), the number of messages could be greater than the given batch size and the remaining messages goes negative, which is why we use an `i64` as we want to keep track of the number of negative messages as the next send flow will be the batch_size - minus remaining messages which allow us to retrieve the casual workflow of flow command messages. 

Here is the example of it works for a batch_size at 1000 and the error case: 

```
Message (1) -> (batch_size = 1000, remaing_messages = 999, no flow message trigger)
... 499 messages later ...
Message (1) -> (batch_size = 1000, remaing_messages = 499, flow message trigger, ask broker to send => batch_size - remaining_messages = 501)
... 200 messages later ...
BatchMessage (1024) -> (batch_size = 1000, remaining_messages = 4294967096, no flow message trigger) [underflow on remaining messages - without the patch]
BatchMessage (1024) -> (batch_size = 1000, remaining_messages = -1124, flow message trigger, ask broker to send =>  batch_size - remaining_messages = 2124) [no underflow on remaining messages - with the patch]
```